pkg/image/prune/imagepruner.go

```diff
@@ -53,12 +67,59 @@ type ImagePruner interface {
 
 // imagePruner implements ImagePruner.
 type imagePruner struct {
-	g         graph.Graph
-	algorithm pruneAlgorithm
+	g              graph.Graph
+	algorithm      pruneAlgorithm
+	registryPinger registryPinger
 }
 
 var _ ImagePruner = &imagePruner{}
 
+// registryPinger performs a health check against a registry.
+type registryPinger interface {
+	// ping performs a health check against registry.
+	ping(registry string) error
+}
+
+// defaultRegistryPinger implements registryPinger.
+type defaultRegistryPinger struct {
+}
+
+func (drp *defaultRegistryPinger) ping(registry string) error {
+	healthzCheck := func(proto, registry string) error {
+		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
```

Comment:
Yeah, the registry component owns ensuring unreferenced content from
manifests is cleaned up.  It's the responsibility of OpenShift to ensure
manifests not backed by an image stream are cleaned up.

On Tue, Jun 30, 2015 at 4:11 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/image/prune/imagepruner.go
> <https://github.com/openshift/origin/pull/3515#discussion_r33617836>:
>
> >  }
> >
> >  var _ ImagePruner = &imagePruner{}
> >
> > +// registryPinger performs a health check against a registry.
> > +type registryPinger interface {
> > +	// ping performs a health check against registry.
> > +	ping(registry string) error
> > +}
> > +
> > +// defaultRegistryPinger implements registryPinger.
> > +type defaultRegistryPinger struct {
> > +}
> > +
> > +func (drp *defaultRegistryPinger) ping(registry string) error {
> > +	healthzCheck := func(proto, registry string) error {
> > +		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
>
> This is another reason why it'd be nicer to have the registry drive
> deleting its own resources...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3515/files#r33617836>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift